### ISSUE TITLE
DB crash

### ISSUE THREAD

**Issue by ryanpitt** [suffolk-northern/main#32]
Something about the database is locked.

Edit: Not necessarily "locked", now I'm seeing the DB is closed and `NullPointerException`.

**Comment by ryanpitt** [comment]
```
Apr 03, 2018 3:24:37 PM track_model.DbHelper query
SEVERE: null
java.sql.SQLException: database connection closed
	at org.sqlite.core.CoreConnection.checkOpen(CoreConnection.java:336)
	at org.sqlite.jdbc4.JDBC4Connection.createStatement(JDBC4Connection.java:38)
	at org.sqlite.jdbc3.JDBC3Connection.createStatement(JDBC3Connection.java:193)
	at track_model.DbHelper.query(DbHelper.java:58)
	at track_model.TrackModel.getBlock(TrackModel.java:127)
	at track_model.TrackModel.getClosestBlock(TrackModel.java:492)
	at track_model.TrackModel.update(TrackModel.java:538)
	at updater.Updater.iteration(Updater.java:49)
	at updater.Updater$Task.run(Updater.java:73)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)

Exception in thread "Timer-0" java.lang.NullPointerException
	at track_model.TrackModel.getBlock(TrackModel.java:128)
	at track_model.TrackModel.getClosestBlock(TrackModel.java:492)
	at track_model.TrackModel.update(TrackModel.java:538)
	at updater.Updater.iteration(Updater.java:49)
	at updater.Updater$Task.run(Updater.java:73)
	at java.util.TimerThread.mainLoop(Timer.java:555)
	at java.util.TimerThread.run(Timer.java:505)
```

Steps to reproduce:
 1. Run `integration.Main` when a database already exists.
 2. Open and close the track model UI.

**Comment by rogxue** [comment]
Fixed with substantial database changes, will need to test on other machines but works on mine

**Comment by ryanpitt** [comment]
Resolved by #41.
